app/mailers/patient_mailer.rb

```diff
@@ -108,14 +114,13 @@ def assessment_sms(patient)
                max_retries_message: I18n.t('assessments.sms.prompt.max_retries_message', locale: lang),
                thanks: I18n.t('assessments.sms.prompt.thanks', locale: lang) }
 
+    update_last_assessment_reminder_sent(patient) # Update last send attempt timestamp before Twilio call
     if TwilioSender.start_studio_flow(patient, params)
       add_success_history(patient, patient)
     else
       add_fail_history_sms(patient)
+      clear_last_assessment_reminder_sent(patient) # Reset send attempt timestamp on failure
     end
-
-    # Always update the last contact time so the system does not try and send emails again.
-    patient.update(last_assessment_reminder_sent: DateTime.now)
```

Comment:
So this is obviously different functionality from what we explicitly called out in the old comment here, and I want to lay out the cases to consider to better understand why we originally had opposite functionality to this change (setting `last_assessment_reminder_sent` even on failure).

1. Fails due to a blocked number 
2. Fails due to an invalid number 
3. Fails due to Twilio execution already existing
4. Fails due to other unexpected Twilio issue such as rate limiting

1 - 3 are valid reasons to _not_ retry, and with this change we will try them again each hour of their monitoring period. 4 _is_ something we would want to retry on, so the change is positive in this case. 

Are we okay with retrying in 1-3 above when we don't really need to? @mayerm94 Would like your thoughts here in case I'm not thinking of something else that originally warranted this.

Reply:
It's worth mentioning that email now works the same as sms/phone in the context of preferred contact times - so before every time the send job was run (24 times a day), technically an email could be sent. Since that is no longer true, this logic should act more like the others.